Runtime/Triangulator.cs

```diff
@@ -173,187 +234,217 @@ public class InputData
-            dependencies = Settings.RefineMesh ? new RefineMeshJob(this).Schedule(dependencies) : dependencies;
-
-            dependencies = Settings.Preprocessor switch
-            {
-                Preprocessor.PCA => SchedulePCAInverseTransformation(dependencies),
-                Preprocessor.COM => ScheduleLocalToWorldTransformation(dependencies),
-                Preprocessor.None => dependencies,
-                _ => throw new NotImplementedException()
-            };
-
-            return dependencies;
-        }
-
-        private JobHandle SchedulePCATransformation(JobHandle dependencies)
-        {
-            tmpInputPositions = Input.Positions;
-            Input.Positions = localPositions.AsDeferredJobArray();
-            if (Input.HoleSeeds.IsCreated)
-            {
-                tmpInputHoleSeeds = Input.HoleSeeds;
-                Input.HoleSeeds = localHoleSeeds.AsDeferredJobArray();
-            }
-
-            dependencies = new PCATransformationJob(this).Schedule(dependencies);
-            if (tmpInputHoleSeeds.IsCreated)
-            {
-                dependencies = new PCATransformationHolesJob(this).Schedule(dependencies);
-            }
-            return dependencies;
-        }
-
-        private JobHandle SchedulePCAInverseTransformation(JobHandle dependencies)
-        {
-            dependencies = new PCAInverseTransformationJob(this).Schedule(this, dependencies);
-
-            Input.Positions = tmpInputPositions;
-            tmpInputPositions = default;
-
-            if (tmpInputHoleSeeds.IsCreated)
-            {
-                Input.HoleSeeds = tmpInputHoleSeeds;
-                tmpInputHoleSeeds = default;
-            }
-
-            return dependencies;
-        }
-
-        private JobHandle ScheduleWorldToLocalTransformation(JobHandle dependencies)
-        {
-            tmpInputPositions = Input.Positions;
-            Input.Positions = localPositions.AsDeferredJobArray();
-            if (Input.HoleSeeds.IsCreated)
-            {
-                tmpInputHoleSeeds = Input.HoleSeeds;
-                Input.HoleSeeds = localHoleSeeds.AsDeferredJobArray();
-            }
-
-            dependencies = new InitialLocalTransformationJob(this).Schedule(dependencies);
-            if (tmpInputHoleSeeds.IsCreated)
-            {
-                dependencies = new CalculateLocalHoleSeedsJob(this).Schedule(dependencies);
+        [BurstCompile]
+        struct TriangulationJob: IJob {
+            public Preprocessor preprocessor;
+            public bool validateInput;
+            public bool restoreBoundary;
+            public bool refineMesh;
+            public int sloanMaxIters;
+            public float concentricShellsParameter;
+            public float refinementThresholdArea;
+            public float refinementThresholdAngle;
```

Comment:
Same here.

You can add constructor 

```csharp
public TriangulationJob(Triangulator triangulator)
{
...
}
```

and make all `int/float/bool` fields 

```csharp
private readonly int/float/bool
```